sys/net/gnrc/routing/rpl/gnrc_rpl_validation.c

```diff
@@ -0,0 +1,178 @@
+/*
+ * Copyright (C) 2016 Cenk Gündoğan <mail@cgundogan.de>
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser General
+ * Public License v2.1. See the file LICENSE in the top level directory for more
+ * details.
+ */
+
+/**
+ * @{
+ *
+ * @file
+ *
+ * @author  Cenk Gündoğan <mail@cgundogan.de>
+ */
+
+#include "net/gnrc/icmpv6.h"
+#include "net/gnrc/ipv6.h"
+
+#include "net/gnrc/rpl.h"
+#include "gnrc_rpl_internal/validation.h"
+
+#define ENABLE_DEBUG    (0)
+#include "debug.h"
+
+#if ENABLE_DEBUG
+static char addr_str[IPV6_ADDR_MAX_STR_LEN];
+#endif
+
+bool gnrc_rpl_validation_DIS(gnrc_rpl_dis_t *dis, uint16_t len)
```

Comment:
had only a very brief look yet, @cgundogan what do you think about `inline` all these functions?